src/PatchOrchestrationApplication/NodeAgentService/src/NTServiceConfigurationUtility.cs

```diff
@@ -35,6 +38,76 @@ internal static void CreateConfigurationForNtService(ConfigurationPackage packag
+            if (configurationSection != null)
+            {
+
+                string paramName = "WUOperationRetryCount";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<long>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+
+                paramName = "WUDelayBetweenRetriesInMinutes";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<long>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "WUOperationTimeOutInMinutes";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<long>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "WURescheduleTimeInMinutes";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<long>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "WURescheduleCount";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<long>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "DisableWindowsUpdates";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<bool>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "OperationTimeOutInMinutes";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<long>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "InstallWindowsOSOnlyUpdates";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<bool>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+                paramName = "AcceptWindowsUpdateEula";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<bool>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+            }
+        } 
+
+
+        private static void ValidateParameter<T> (string paramName, string value, FabricClient fabricClient, ServiceContext serviceContext)
```

Comment:
>ValidateParameter [](start = 28, length = 17)

Where are we validating the WU frequency? That is the most common setting people modify and add incorrectly. Either specify days as 29 or incorrect format.

Reply:
It is validated at NodeAgentNTService. It was a big code, so it did not make much sense to replicate it at NodeAgentService. But if someone inputs incorrect value for WUFrequency, we will post warning on the service.

Reply:
Where are we posting that warning?

---
In reply to: [282805214](https://github.com/microsoft/Service-Fabric-POA/pull/16#discussion_r282805214) [](ancestors = 282805214)